src/cirrus/plugins/management/deployment.py

```diff
@@ -187,16 +194,14 @@ def del_user_var(self, name, save=False):
             self.save()
 
     def exec(self, command, include_user_vars=True, isolated=False):
-        import os
-
         if isolated:
             env = self.environment.copy()
             if include_user_vars:
                 env.update(self.user_vars)
-            os.execlpe(command[0], *command, env)
-
-        self.set_env(include_user_vars=include_user_vars)
-        os.execlp(command[0], *command)
+            check_call(command, env=env)
+        else:
+            self.set_env(include_user_vars=include_user_vars)
+            check_call(command)
```

Comment:
Are you dependent this change elsewhere? The choice to use an exec here vs a subprocess call was specifically because it is used by the cli with the intent to run another command with a specific context, which to me is very different than code wanting to shell out to run something.

In short, I think my suggestion is that we don't change the `exec` implementation, but instead add another method (like `call` maybe?) that uses subprocess for where that behavior is required.

Reply:
`call` sounds like a reasonable way to go.  The piece here is to be able to invoke a deployed lambda, using the cirrus-mgmt managed env vars, and then collect its output within pytest.  `os.execlp` hands over the whole process to the command, which exits when the aws cli exits, and cannot collect the output.

I suppose another approach would be to add a `run_lambda` method, and use that in a fashion similar to `run_workflow`.  Not sure if that'd be cleaner.

Reply:
I think a new `call` method makes sense to me, if it works for what need. A `run_lambda` method could also be advantageous for the scenario you are proposing, as you could implement it without shelling out to the aws cli, but I'm not sure it would be as generally useful as `call`.